rust/kernel/sysctl.rs

```diff
@@ -74,13 +72,14 @@ impl SysctlStorage for atomic::AtomicBool {
         (data.len(), result)
     }
 
-    fn read_value(&self, data: &mut UserSlicePtrWriter) -> (usize, Result) {
+    fn read_value(&self, data: &mut [u8]) -> (usize, Result) {
         let value = if self.load(atomic::Ordering::Relaxed) {
             b"1\n"
         } else {
             b"0\n"
         };
-        (value.len(), data.write_slice(value))
+        let dst = &mut data[..value.len()];
```

Comment:
What if `data` is a zero-sized buffer? This would panic.

Reply:
Thanks for the quick feedback, you’re totally right of course!